ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/request/heal/AccountTrieNodeHealingRequest.java

```diff
@@ -112,6 +112,9 @@ public Stream<SnapDataRequest> getRootStorageRequests(final WorldStateStorage wo
                   account.size() - getLocation().size()))
           .map(RLP::input)
           .map(StateTrieAccountValue::readFrom)
+          .filter(
+              stateTrieAccountValue ->
+                  !stateTrieAccountValue.getStorageRoot().equals(MerkleTrie.EMPTY_TRIE_NODE_HASH))
```

Comment:
maybe add a comment here since this is the crucial change
don't request accounts with empty storage since this indicates they can't be healed

Reply:
added